pkg/kubefedctl/enable/validation.go

```diff
@@ -243,6 +243,14 @@ func ValidationSchema(specProps v1beta1.JSONSchemaProps) *v1beta1.CustomResource
 					},
 				},
 			},
+			// Require a spec (even if empty) as an aid to users
+			// manually creating federated configmaps or
+			// secrets. These target types do not include a spec,
+			// and the absence of the spec in a federated
+			// equivalent could indicate a malformed resource.
```

Comment:
Wouldn't the lack of `spec` in a federated equivalent always indicate a malformed resource? Otherwise there'd be no way to provide overrides, placement, etc.

Reply:
Not necessarily - overrides are entirely optional, the absence of placement indicates that the resource should not exist in any cluster, and namespaces don't even require a template.